### PR TITLE
[terra-list] Enable navigation to the list items with arrow keys.

### PR DESCRIPTION
**Pull request by AH106586Harika**
### Summary
 Fixed : Enable navigation to the list items with arrow keys.

**What was changed:**
Added cyclic navigation using Keyboard with arrow keys for the list-items

**Why it was changed:**
To implement cyclic navigation between list items with arrow keys.

### Testing

This change was tested using:

- [X] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [x] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9644 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-list/src/ListSection.jsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
```
Comment (supreethmr) [inline packages/terra-list/src/ListSection.jsx:1]: this should not be added for whole file instead you can add it above the statement that causing lint error.

File: packages/terra-core-docs/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* Added
+  * Added cyclic navigation to support keyboard usage with arrow keys.
```
Comment (supreethmr) [inline packages/terra-core-docs/CHANGELOG.md:6]:
```suggestion
* Changed
  * Updated terra-list examples to demonstrate arrow key navigation within list and list sections.
```

File: packages/terra-list/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 ## Unreleased
 * Added
+  * Added cyclic navigation to support keyboard usage with arrow keys.
```
Comment (supreethmr) [inline packages/terra-list/CHANGELOG.md:5]:
```suggestion
  * Added arrow key navigation support for list items.
```
Please add space between Unreleased and Added section

Comment (MadanKumarGovindaswamy) [reply packages/terra-list/CHANGELOG.md:5]: Changes done. Thanks

File: packages/terra-core-docs/CHANGELOG.md (hunk 2 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-core-docs/CHANGELOG.md:6]: Changes done. Thanks

File: packages/terra-list/src/ListSubsection.jsx [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
```
[the remark this answers is not in the document]
Comment (MadanKumarGovindaswamy) [reply packages/terra-list/src/ListSubsection.jsx:1]: Changes done. Thanks

File: packages/terra-list/src/ListSection.jsx (hunk 1 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-list/src/ListSection.jsx:1]: Changes done. Thanks

File: packages/terra-list/src/List.jsx [hunk 5]
```diff
@@ -147,14 +147,23 @@ const List = ({
...
       listItems[0].focus();
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_UP) {
+      event.preventDefault();
+      const previousIndex = currentIndex > 0 ? currentIndex - 1 : lastIndex;
+      listItems[previousIndex].focus();
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_DOWN) {
+      event.preventDefault();
+      const nextIndex = currentIndex < lastIndex ? currentIndex + 1 : 0;
+      listItems[nextIndex].focus();
     }
```
Comment (sdadn) [inline packages/terra-list/src/List.jsx:167]: A switch would be more suitable than a if/else ladder.

**Review by sdadn — CHANGES_REQUESTED** [review]

File: packages/terra-list/src/List.jsx (hunk 5 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-list/src/List.jsx]: changes done Thanks

**Review by supreethmr — APPROVED** [review]

File: packages/terra-list/CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,7 @@
 ## 4.63.0 - (September 26, 2023)
 
 * Added
+  * Added arrow key navigation support for list items.
```
Comment (supreethmr) [inline packages/terra-list/CHANGELOG.md:8]: this should be moved to unreleased section !!

Comment (MadanKumarGovindaswamy) [reply packages/terra-list/CHANGELOG.md]: Added thanks

**Pull request merged**